### PR TITLE
Allow httpd Services, Deployment and ConfigMap to be reconciled

### PR DESCRIPTION
**Pull request by bdunne**
Applies to #438

### CODE REVIEW

File: manageiq-operator/pkg/helpers/miq-components/httpd.go [hunk 1]
```diff
@@ -278,197 +303,195 @@ func initializeHttpdContainer(spec *miqv1alpha1.ManageIQSpec, privileged bool, c
...
-							Name: "httpd-config",
-							VolumeSource: corev1.VolumeSource{
-								ConfigMap: &corev1.ConfigMapVolumeSource{
-									LocalObjectReference: corev1.LocalObjectReference{Name: "httpd-configs"},
-								},
-							},
-						},
-					},
 				},
+				Spec: corev1.PodSpec{},
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/httpd.go:337]: Same with this, why add it here?

File: manageiq-operator/pkg/helpers/miq-components/httpd.go [hunk 2]
```diff
@@ -278,197 +303,195 @@ func initializeHttpdContainer(spec *miqv1alpha1.ManageIQSpec, privileged bool, c
...
-		"app": cr.Spec.AppName,
-	}
-	selector := map[string]string{
-		"service": "web-service",
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, service, scheme); err != nil {
+			return err
+		}
+		addAppLabel(cr.Spec.AppName, &service.ObjectMeta)
+		service.Spec.Ports[0].Name = "ui-service-3000"
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/httpd.go:388]: Same as https://github.com/ManageIQ/manageiq-pods/pull/577/files#r450876638, I think we need to check for nil as that's the array zero value.

I'm not sure if the `Get` in `CreateOrUpdate` will overwrite the Ports array, but it's probably safer to remove it from the spec above and `make` a new one here if it's nil

Comment (carbonin) [reply manageiq-operator/pkg/helpers/miq-components/httpd.go:388]: This really applies to all of the other services as well. I'll only write it here though.

File: manageiq-operator/pkg/helpers/miq-components/httpd.go [hunk 3]
```diff
@@ -278,197 +303,195 @@ func initializeHttpdContainer(spec *miqv1alpha1.ManageIQSpec, privileged bool, c
...
+				Spec: corev1.PodSpec{},
 			},
 		},
 	}
 
-	configureHttpdAuth(&cr.Spec, &deployment.Spec.Template.Spec)
-
 	// Only assign the service account if we need additional privileges
 	if privileged {
 		deployment.Spec.Template.Spec.ServiceAccountName = cr.Spec.AppName + "-httpd"
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/httpd.go:344]: This should be in the mutate function, right?

File: manageiq-operator/pkg/helpers/miq-components/httpd.go [hunk 4]
```diff
@@ -8,97 +8,122 @@ import (
...
-					},
-					SecretName: tlsSecretName(cr),
+		Spec: extensionsv1beta1.IngressSpec{},
+	}
+
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, ingress, scheme); err != nil {
+			return err
+		}
+		ingress.Spec.TLS = []extensionsv1beta1.IngressTLS{
```
[the remark this answers is not in the document]
Comment (bdunne) [reply manageiq-operator/pkg/helpers/miq-components/httpd.go]: Done

File: manageiq-operator/pkg/helpers/miq-components/httpd.go (hunk 2 above)
Comment (bdunne) [reply manageiq-operator/pkg/helpers/miq-components/httpd.go:387]: Done

File: manageiq-operator/pkg/helpers/miq-components/httpd.go (hunk 3 above)
Comment (bdunne) [reply manageiq-operator/pkg/helpers/miq-components/httpd.go]: Done

File: manageiq-operator/pkg/helpers/miq-components/httpd.go [hunk 5]
```diff
@@ -278,197 +302,195 @@ func initializeHttpdContainer(spec *miqv1alpha1.ManageIQSpec, privileged bool, c
...
+	f := func() error {
+		if err := controllerutil.SetControllerReference(cr, service, scheme); err != nil {
+			return err
+		}
+		addAppLabel(cr.Spec.AppName, &service.ObjectMeta)
+		if len(service.Spec.Ports) == 0 {
+			service.Spec.Ports = append(service.Spec.Ports, corev1.ServicePort{})
+		}
+		service.Spec.Ports[0].Name = "remote-console-3000"
+		service.Spec.Ports[0].Port = 3000
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/httpd.go:438]: :+1:

**Comment by bdunne** [comment]
> This actually fixes #438, right?

Yes it does

**Pull request merged**